src/commands/notify.rs

```diff
@@ -142,20 +109,111 @@ pub struct Notify {
 
     /// Max number of retries
     #[arg(short = 'r', long = "retries", default_value = "15")]
-    retries: usize,
+    retries: u8,
 
     /// DNS servers to send packet to
     #[arg(required = true)]
     servers: Vec<String>,
 }
 
+const LDNS_HELP: &str = "\
+usage: ldns-notify [other options] -z zone <servers>
+Ldns notify utility
+
+ Supported options:
+        -z zone         The zone
+        -I <address>    source address to query from
+        -s version      SOA version number to include
+        -y <name:key[:algo]>    specify named base64 tsig key, and optional an
+                        algorithm (defaults to hmac-md5.sig-alg.reg.int)
+        -p port         port to use to send to
+        -v              Print version information
+        -d              Print verbose debug information
+        -r num          max number of retries (15)
+        -h              Print this help information
+
+Report bugs to <dns-team@nlnetlabs.nl>\
+";
+
+impl LdnsCommand for Notify {
+    const NAME: &'static str = "notify";
+    const HELP: &'static str = LDNS_HELP;
+    const COMPATIBLE_VERSION: &'static str = "1.8.4";
+
+    fn parse_ldns<I: IntoIterator<Item = std::ffi::OsString>>(args: I) -> Result<Args, Error> {
+        let mut zone = None;
+        let mut soa_version = None;
+        let mut tsig = None;
+        let mut port = 53;
+        let mut debug = false;
+        let mut retries = 15;
+        let mut servers = Vec::new();
+
+        let mut parser = lexopt::Parser::from_args(args);
+
+        while let Some(arg) = parser.next()? {
+            match arg {
+                Arg::Short('z') => {
+                    let val = parser.value()?;
+                    zone = Some(parse_os("zone (-z)", &val)?);
+                }
+                Arg::Short('I') => todo!(),
```

Comment:
Maybe this should be an error instead saying it's currently unsupported?

Reply:
You're absolutely right of course!